server/routerlicious/packages/services-shared/src/configDumper.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fastRedact from "fast-redact";
+import { ILogger } from "@fluidframework/server-services-core";
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+const errorSanitizationMessage = "FluidREDACTED";
+
+export class ConfigDumper {
+	private readonly config: Record<string, any>;
+	private readonly secretNamesToRedactInConfigDump = [
+		"mongo.globalDbEndpoint",
+		"mongo.operationsDbEndpoint",
+		"redis.pass",
+		"redisForTenantCache.pass",
+		"redis2.pass",
+		"redisForThrottling.pass",
+	];
+	private readonly logger: ILogger | undefined;
+
+	// Check library for issues/malware
+	constructor(
+		config: Record<string, any>,
+		logger: ILogger | undefined,
+		secretNamesToRedactInConfigDump?: string[],
+	) {
+		this.config = JSON.parse(JSON.stringify(config));
+		if (secretNamesToRedactInConfigDump !== undefined) {
+			this.secretNamesToRedactInConfigDump = this.secretNamesToRedactInConfigDump.concat(
+				secretNamesToRedactInConfigDump,
+			);
+		}
+		this.logger = logger;
+	}
+
+	public dumpConfig() {
+		const redactJsonKeys = fastRedact({
+			paths: this.secretNamesToRedactInConfigDump,
+			censor: errorSanitizationMessage,
+			serialize: false,
+		});
+
+		try {
+			redactJsonKeys(this.config);
```

Comment:
Juan is working on adding logs sanitization in the form of formatters plug-in for Lumberjack. There seems to be some overlap between these two tasks. In case you switch to using Lumberjack for logging you should be able to leverage Juan's work.

Reply:
This is a good point. I initially thought the two tasks were similar but would not overlap, but perhaps we can discuss how to leverage each other's work here if that is applicable.

One more point for the discussion: the current sanitization logic only looks at the exception field of Lumberjack, and not on properties/message, so Dhruv would have to add a new formatter. At the same time, we do not want that formatter to be executing on the properties/message for each service/cluster instance (at least that is how I understand the sanitization logic work - to detect secret leaks in logs in early deployment rings), while this feature Dhruv is working on would be enabled everywhere

Reply:
We can use the LogLevel.Debug or maybe introduce a new level i.e. LogLevel.Config for this purpose ?

Reply:
This would only be logged once per service on startup, so it should not contribute to many logs/noise. Debug level would also mean we don't capture this information in all environments, since most only allow Info level and above.